front-end/src/components/RollCallComponents/GroupsComponent.vue

```diff
@@ -0,0 +1,128 @@
+<template>
+	<div class="grid grid-cols-3 gap-8 mx-auto w-3/4 max-w-2x h-100">
+		<div
+			v-for="(group, index) in groups"
+			:key="index"
+			class="bg-white text-gray-500 shadow rounded-lg"
+		>
+			<h2 class="bg-gray-100 p-2 rounded-t-lg font-bold text-center">Group {{ index + 1 }}</h2>
+			<div
+				v-for="(student, sIndex) in group"
+				:key="sIndex"
+				class="flex flex-row items-center p-2 border-b"
+			>
+				<div class="w-1/3">{{ sIndex + 1 }}</div>
+				<div class="w-2/3">
+					{{
+						student
+							? student.profile.firstName + ' ' + student.profile.lastName
+							: 'Rejoindre le groupe'
+					}}
+				</div>
+			</div>
+			<div class="flex justify-center items-center p-4">
+				<button
+					class="py-2 px-4 bg-blue-500 hover:bg-blue-700 text-white font-bold rounded"
+					@click="joinGroup(index)"
+				>
+					Join Group {{ index + 1 }}
+				</button>
+			</div>
+		</div>
+	</div>
+	<div v-if="isPO" class="flex justify-center p-4 space-x-4">
+		<button
+			class="py-2 px-4 bg-blue-500 hover:bg-blue-700 text-white font-bold rounded"
+			@click="createRandomGroups"
+		>
+			Create Random Groups
+		</button>
+		<button
+			class="py-2 px-4 bg-blue-500 hover:bg-blue-700 text-white font-bold rounded"
+			@click="emptyGroups"
+		>
+			Empty Groups
+		</button>
+	</div>
+</template>
+
+<script lang="ts">
+import { http } from '@/api/network/axios';
+import { withErrorHandler } from '@/utils/storeHandler';
+
+let courseId = '';
```

Comment:
Same, i'm not i understand the necessity of having them both inside and outside.
Also, those outside are definitely not reactive.